cvat-sdk/cvat_sdk/auto_annotation/driver.py

```diff
@@ -16,13 +16,39 @@
 from cvat_sdk.core.progress import NullProgressReporter, ProgressReporter
 from cvat_sdk.datasets.task_dataset import TaskDataset
 
+from ..attributes import attribute_value_validator
 from .exceptions import BadFunctionError
 from .interface import DetectionFunction, DetectionFunctionContext, DetectionFunctionSpec
 
 
+@attrs.frozen
+class _AttributeNameMapping:
```

Comment:
Shouldn't these classes called `*Mapping` be `collections.abc.Mapping`-compatible?

Reply:
Well... it's a more abstract notion of mapping. The idea is that each of the `XNameMapping` classes defines how names within `X` should be mapped, so it parallels the structure of `X`. So, for example, if you want to map `attribute.name`, you look up `attribute_nm.name` (where `attribute_nm` is an `_AttributeNameMapping` object).

Maybe they shouldn't be called `*Mapping`, but I couldn't think of anything better when I created them.